contracts/adapters/IPRebalanceExtension.sol

```diff
@@ -0,0 +1,555 @@
+     */
+    constructor(
+        IBaseManager _manager,
+        IGeneralIndexModule _generalIndexModule,
+        IAirdropModule _airdropModule
+    )
+        public
+        GIMExtension(_manager, _generalIndexModule)
+    {
+        airdropModule = _airdropModule;
+    }
+
+    /* ======== External Functions ======== */
+
+    /**
+     * Original function for starting rebalances from the inherited GIMExtension. This function has been deprecated
+     * in favor of startIPRebalance. It will revert when it is called.
+     */
+    function startRebalanceWithUnits(
+        address[] calldata /* _components */,
+        uint256[] calldata /* _targetUnitsUnderlying */,
+        uint256 /* _positionMultiplier */
+    )
+        external
+        onlyOperator
+        override
+    {
+        revert("use startIPRebalance instead");
+    }
+
+    /**
+     * ONLY OPERATOR: Sets a TransformInfo entry for a transform component. Each transform component must have a TransformInfo set
+     * for it which includes neccessary information such as the underlying component and TransformHelper address. Must not already
+     * have been set. If it has, use updateTransformInfo.
+     *
+     * @param _transformComponent       component that needs to be transformed/untransformed during rebalance
+     * @param _transformInfo            TransfomInfo entry for the transform component
+     */
+    function setTransformInfo(address _transformComponent, TransformInfo memory _transformInfo) external onlyOperator {
+        require(
+            transformComponentInfo[_transformComponent].underlyingComponent == address(0),
+            "TransformInfo already set"
+        );
+        
+        // Maximum transform size is limited to MAX_UINT_96 to prevent overflows when multiplying this maximum size by
+        // the transform components exchange rate to get a maximum untransform size. This prevents using MAX_UINT_256 
+        // to represent an unlimited amount.
+        require(_transformInfo.maxTransformSize <= type(uint96).max, "max transform size must be less than MAX_UINT_96");
+        transformComponentInfo[_transformComponent] = _transformInfo;
+    }
+
+    /**
+     * ONLY OPERATOR: Updates TransformInfo entry for a transform component. Must already have been set before updating. If not,
+     * use setTransformInfo.
+     *
+     * @param _transformComponent       component that needs to be transformed/untransformed during rebalance
+     * @param _transformInfo            new TransfomInfo entry for the transform component
+     */
+    function updateTransformInfo(address _transformComponent, TransformInfo memory _transformInfo) external onlyOperator {
+        require(
+            transformComponentInfo[_transformComponent].underlyingComponent != address(0),
+            "TransformInfo not set yet"
+        );
+
+        // Maximum transform size is limited to MAX_UINT_96 to prevent overflows when multiplying this maximum size by
```

Comment:
We have the same piece of code in this and the above function. We can add it in an internal function instead.